docs/rules-engine/how-to-use/built-in-facts.md

```diff
@@ -0,0 +1,114 @@
+# Rules engine - Built-in facts
+
+In some cases, the Otter [operators](../README.md#operator) may refer to an implicit fact. This is the case for
+the `dateInNextMinutes` and `dateNotInNextMinutes` operators.
+
+The definition of these two operators implies the notion of a current date, hence, the framework needs to expose an 
+internal, built-in fact that will always be used by them: the `o3rCurrentTime`. The operators will not take it as input, 
+making it an implicit fact of these operators. This also means that, while all the other operators are pure functions,
+this is not the case for `dateInNextMinutes` nor `dateNotInNextMinutes`.
+
+While this fact is provided in the rules engine module, you should be able to override its implementation to fit your needs.
+
+This section explains how to use operators with an implicit fact and how to override the fact implementation.
+
+## Usage
+The operators using a built-in implicit fact are used in the same way as any other operator, the implicit fact will just
+not be referred to in the Ruleset object:
+```json55
+{
+  "id": "5467e501-b9ff-414f-8026-56885d0d7a4c",
+  "name": "The otter is late",
+  "disabled": false,
+  "outputRuntimeFacts": [],
+  "inputRuntimeFacts": [],
+  "rootElement": {
+    "elementType": "RULE_BLOCK",
+    "blockType": "IF_ELSE",
+    "condition": {
+      "all": [
+        {
+          "lhs": {
+            "type": "FACT",
+            "value": "outboundDate"
+          },
+          "rhs": {
+              "type": "LITERAL",
+              "value": "2880"
+          },
+          "operator": "dateInNextMinutes"
+        }
+      ]
+    },
+    "successElements": [
+      {
+        "elementType": "ACTION",
+        "actionType": "UPDATE_ASSET",
+        "asset": "otter.svg",
+        "value": "otter-late.svg"
+      }
+    ],
+    "failureElements": []
+  }
+}
+```
+
+This does not mean that the fact itself should not be registered.
+If you fail to register the fact the same way as the non-built-in facts, the whole ruleset will fail.
+
+```typescript
+import {Component, OnInit} from '@angular/core';
+import {CurrentTimeFactsService} from '@o3r/rules-engine';
+
+@Component({
+  selector: 'app-root',
+  templateUrl: './app.component.html',
+})
+export class AppComponent implements OnInit {
+  constructor(private currentTimeFactsService: CurrentTimeFactsService) {}
+
+  ngOnInit() {
+    this.currentTimeFactsService.register();
+  }
+}
+```
+
+The ``CurrentTimeFactsService`` is based on a simple implementation of a refresh mechanism. It is up to the 
```

Comment:
and we do not provide a default refresh timer to avoid any performance issue. Instead, it is up to...